From 01348b2a9de65b400f45e2b36829e21d3a2378f7 Mon Sep 17 00:00:00 2001
From: Vidhya Vidhyaraman <vraman@marvell.com>
Date: Tue, 28 May 2019 14:21:23 -0700
Subject: [PATCH 259/386] octeontx: Fix LBK1/2 channel index

This commit fixes the incorrect index
of LBK1/2 channel used during lbk1/2 port start.
Also fixed incorrect conditional that checks for
lbk_mode not being set when packet is sent from kernel
netdevice lbk0 to ODP.

Change-Id: I241798aa0c92169737111ccc6598b6082aa9da8a
Signed-off-by: Vidhya Vidhyaraman <vraman@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/10128
Reviewed-by: Yuri Tolstov <Yuri.Tolstov@cavium.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/net/ethernet/cavium/octeontx-83xx/lbk_main.c | 5 ++++-
 drivers/net/ethernet/cavium/thunder/nicvf_main.c     | 2 +-
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/cavium/octeontx-83xx/lbk_main.c b/drivers/net/ethernet/cavium/octeontx-83xx/lbk_main.c
index 956571aad2aa..3fd5391f9690 100644
--- a/drivers/net/ethernet/cavium/octeontx-83xx/lbk_main.c
+++ b/drivers/net/ethernet/cavium/octeontx-83xx/lbk_main.c
@@ -268,10 +268,13 @@ int lbk_port_start(struct octtx_lbk_port *port)
 		if (rc)
 			return -EIO;
 		pkind = thlbk->get_port_pkind();
+		i = 0; /* LBK channel to set PKIND for LBK1/2 port */
 	} else {
 		pkind = port->pkind;
+		/* LBK channel to set PKIND for LBK0 port */
+		i = port->glb_port_idx;
 	}
-	i = port->glb_port_idx;
+
 	lbk = get_lbk_dev(port->node, port->ilbk);
 	lbk_reg_write(lbk, LBK_CH_PKIND(i), port->pkind);
 	lbk = get_lbk_dev(port->node, port->olbk);
diff --git a/drivers/net/ethernet/cavium/thunder/nicvf_main.c b/drivers/net/ethernet/cavium/thunder/nicvf_main.c
index 37832c4e61fa..9dbccd2223f8 100644
--- a/drivers/net/ethernet/cavium/thunder/nicvf_main.c
+++ b/drivers/net/ethernet/cavium/thunder/nicvf_main.c
@@ -1365,7 +1365,7 @@ static netdev_tx_t nicvf_xmit(struct sk_buff *skb, struct net_device *netdev)
 		return NETDEV_TX_OK;
 	}
 	/* Reroute packets from ethX, if interface belongs to Dataplane.*/
-	if (!nic->lbk_mode && nic->port_ctx == NIC_PORT_CTX_DATAPLANE) {
+	if (nic->port_ctx == NIC_PORT_CTX_DATAPLANE) {
 		/* Use LBK media */
 		netdev = snd_pkt_reroute(netdev, skb, &nic);
 		if (!netdev) {
-- 
2.17.1

